V5 upgrade

orenbm opened the pull request.

orenbm committed:
- Creating containers for Conjur server & client
- Applying policy and adding a variable
- Extracting host factory token

orenbm committed:
Renamed demo directory

orenbm committed:
Added container that will run ansible

orenbm committed:
Passing HFTOKEN to ansible-runner as environment variable

orenbm committed:
Some refactoring

Commit:
Fix the host_factories request to v5 Remove the Vagrant files Add target container to docker compose

Commit:
Remove the environment varibales from the target

orenbm committed:
Removed container for running ansible

orenbm committed:
Ansible is running and passing

orenbm committed:
Demo for running Ansible on Docker container & fetching Conjur secret inside is running and passing

orenbm commented:
#### What does this PR do?
Add a demo for the ansible fest. In the demo we have containers for Conjur server & client which loads a policy, adds a secret & creates a host factory token. Then, the Ansible runs its playbook on the target container (conjurizing it) and finally we go into the target and show that we fetch the secret. Everything’s pretty clear in `demo/run.sh` but if something’s unclear please ask Jenia or me.


#### Where should the reviewer start?
demo/run.sh

#### How should this be manually tested?
run the `demo/run.sh` file. Note that you need to have Ansible (& Docker) installed on your machine

#### What are the relevant tickets?
#1 

#### Reviewers
@dustinmm80 @doodlesbykumbi @jvanderhoof

jvanderhoof commented on demo/docker-compose.yml:
```diff
@@ -0,0 +1,42 @@
+    container_name: ansible-postgres
+
+  conjur:
+    image: registry.tld/possum:0.1.0-stable
```
Now that we're live on the dockerhub & quay.  We can use those instead of our internal repository:
* `cyberark/conjur:0.1.0-stable`
* `quay.io/cyberark/conjur:0.1.0-stable`

orenbm replied:
+1

orenbm replied:
@nessiCyber will change it in api-java

jvanderhoof commented on demo/docker-compose.yml:
```diff
@@ -0,0 +1,42 @@
+    container_name: ansible-postgres
+
+  conjur:
+    image: registry.tld/possum:0.1.0-stable
```
Now that we're live on the dockerhub & quay.  We can use those instead of our internal repository:
* `cyberark/conjur:0.1.0-stable`
* `quay.io/cyberark/conjur:0.1.0-stable`

doodlesbykumbi commented on .gitignore:
```diff
@@ -2,4 +2,6 @@
-test.sh
+.idea/*
+demo/client-output/*
+*.iml
```
@orenbm i'd suggest newline at EOF. there's a couple other files in the PR without. (https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)

orenbm replied:
+1

doodlesbykumbi commented on demo/run.sh:
```diff
@@ -0,0 +1,88 @@
+
+  createDemoEnvironment
+  waitForServer
+	createHostFactoryToken
```
@orenbm not sure what's happening with this but it seems there are random spacing inconsistencies through this file in particular.

doodlesbykumbi commented on demo/run.sh:
```diff
@@ -0,0 +1,88 @@
+
+	docker exec -e CONJUR_AUTHN_API_KEY=${api_key} \
+		ansible-conjur-client \
+    /bin/bash -c "conjur policy load root /conjurinc/ansible/policies/root.yml && \
```
@orenbm This failed for me citing `/bin/bash: conjurinc/ansible/client-output/hftoken.txt: No such file or directory`.

Still, I would recommend using a bash script here - there's quite a lot happening.

`/bin/bash -c "/conjurinc/ansible/create_host_token.sh"`

In `demo/create_host_token.sh`
```
#!/usr/bin/env bash

cd /conjurinc/ansible
rm -rf client-output
mkdir -p client-output
conjur policy load root policies/root.yml
conjur variable values add password mysecretpassword
export HFTOKEN=$(conjur hostfactory tokens create --duration-days=365 ansible-factory | jq -r '.[0].token')
echo $HFTOKEN > client-output/hftoken.txt

```

orenbm replied:
Great point, looks much better

doodlesbykumbi reviewed (changes requested):
Just a few minor suggestions. Otherwise LGTM

orenbm committed:
- Added newlines to end of text files
- Added script for creating host factory token instead of running as a command
- Pulling Conjur server image from Docker Hub
- Fixed tabs & spaces mix

orenbm committed:
Changed tabs to spaces
